resources/js/common.js

```diff
@@ -283,3 +282,40 @@ function postwith (to,p) {
   document.body.removeChild(myForm) ;
 }
 
+
+function isValidDate(dateString)

```

Comment:
@tuxayo , actually,  no particular test plan nor any integration test plan. The only test by hand plan is set enhancedCostHistory=Y and watch whether all the enhanced fields can  show up in Edit Cost Information and then watch whether Cost information can be saved. Also we need to observe whether those enhanced fields show in Print View which is under the Helpful Links. An extra benefit but not planned is this PR has fixed the error encountered upon saving Cost history. @tuxayo if your test proves fine to you, simply merge it. Thank you.